### PR TITLE
Run amount validation on token/netowork/wallet changes.

### PR DESCRIPTION
**Pull request by nikarm22**
[Ticket](https://www.notion.so/wormholelabs/UI-Bug-Errors-need-to-dynamically-update-based-on-new-inputs-c1dbb35abc3b41aaa198ac20d2fd4117?pvs=23)

### CODE REVIEW

File: wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx [hunk 1]
```diff
@@ -136,19 +131,19 @@ const AmountInput = (props: Props) => {
...
-
       dispatch(setAmount(value));
-      setValidationResult(amountValidation);
     },
     [amount, route, tokenBalance],
   );
 
+  const validationResult = useMemo(
+    () => validateAmount(amount, tokenBalance, getMaxAmt(route)),
+    [amount, tokenBalance, route],
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx:141]: @nikarm22 how is this going to validate on wallet changes?

Comment (nikarm22) [reply wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx:141]: 'tokenBalance' is depending on token, chain and wallet, so it's enough basically.

**Comment by emreboga** [comment]
Closing in favor of https://github.com/wormhole-foundation/wormhole-connect/pull/2551

**Comment by Wesleyleung** [comment]
I'm not sure this issue is solved by #2551? Or at least not fully addressed. The scenario described in the ticket is still reproducible.

**Pull request closed without merging**